src/animation.cpp

```diff
@@ -0,0 +1,429 @@
+#include "game.h"
+#include "debug.h"
+#include "cata_tiles.h" // all animation functions will be pushed out to a cata_tiles function in some manner
+
+extern cata_tiles *tilecontext; // obtained from sdltiles.cpp
+extern void try_update();
+
+// see game.cpp
+bool is_valid_in_w_terrain(int x, int y);
+
+namespace {
+void draw_animation_delay(long const scale = 1)
+{
+    auto const delay = static_cast<long>(OPTIONS["ANIMATION_DELAY"]) * scale * 1000000l;
+
+    timespec const ts = {0, delay};
+    if (ts.tv_nsec > 0) {
+        nanosleep(&ts, nullptr);
+    }
+}
+
+void draw_explosion_curses(WINDOW *const w, int const x, int const y,
+    int const r, nc_color const col)
+{
+    if (r == 0) { // TODO why not always print '*'?
+        mvwputch(w, x, y, col, '*');
+    }
+
+    for (int i = 1; i <= r; ++i) {
+        mvwputch(w, y - i, x - i, col, '/');  // corner: top left
+        mvwputch(w, y - i, x + i, col, '\\'); // corner: top right
+        mvwputch(w, y + i, x - i, col, '\\'); // corner: bottom left
+        mvwputch(w, y + i, x + i, col, '/');  // corner: bottom right
+        for (int j = 1 - i; j < 0 + i; j++) {
+            mvwputch(w, y - i, x + j, col, '-'); // edge: top
+            mvwputch(w, y + i, x + j, col, '-'); // edge: bottom
+            mvwputch(w, y + j, x - i, col, '|'); // edge: left
+            mvwputch(w, y + j, x + i, col, '|'); // edge: right
+        }
+    }
+    
+    wrefresh(w);
+    draw_animation_delay(EXPLOSION_MULTIPLIER);
+}
+} // namespace
+
+#if !defined(SDLTILES)
```

Comment:
This `if( NOT condition ) { A } else { B }` (emphasis on the negation) is a bit confusing. Why did you chose that instead of `if( condition ) { B } else { A }`?